src/status_im2/contexts/chat/lightbox/utils.cljs

```diff
@@ -0,0 +1,164 @@
+  [props state animations]
+  (orientation/use-device-orientation-change
+   (fn [result]
+     (if platform/ios?
+       (handle-orientation result props state animations)
+       ;; `use-device-orientation-change` will always be called on Android, so need to check
+       (orientation/get-auto-rotate-state
+        (fn [enabled?]
+          ;; RNN does not support landscape-right
+          (when (and enabled? (not= result orientation/landscape-right))
+            (handle-orientation result props state animations))))))))
+
+(defn toggle-opacity
+  [index {:keys [opacity-value border-value transparent? props]} portrait?]
+  (let [{:keys [small-list-ref]} props
+        opacity                  (reanimated/get-shared-value opacity-value)]
+    (if (= opacity 1)
+      (do
+        (when platform/ios?
+          ;; status-bar issue: https://github.com/status-im/status-mobile/issues/15343
+          (js/setTimeout #(navigation/merge-options "lightbox" {:statusBar {:visible false}}) 75))
+        (anim/animate opacity-value 0)
+        (js/setTimeout #(reset! transparent? (not @transparent?)) 400))
+      (do
+        (reset! transparent? (not @transparent?))
+        (js/setTimeout #(anim/animate opacity-value 1) 50)
+        (js/setTimeout #(when @small-list-ref
+                          (.scrollToIndex ^js @small-list-ref #js {:animated false :index index}))
+                       100)
+        (when (and platform/ios? portrait?)
+          (js/setTimeout #(navigation/merge-options "lightbox" {:statusBar {:visible true}}) 150))))
+    (anim/animate border-value (if (= opacity 1) 0 12))))
+
+(defn drag-gesture
+  [{:keys [pan-x pan-y background-color opacity layout]} x? set-full-height?]
+  (->
+    (gesture/gesture-pan)
+    (gesture/enabled true)
+    (gesture/max-pointers 1)
+    (gesture/on-start #(reset! set-full-height? false))
+    (gesture/on-update (fn [e]
+                         (let [translation (if x? (oget e "translationX") (oget e "translationY"))
+                               progress    (Math/abs (/ translation constants/drag-threshold))]
+                           (anim/set-val (if x? pan-x pan-y) translation)
+                           (anim/set-val opacity (- 1 progress))
+                           (anim/set-val layout (* progress -20)))))
+    (gesture/on-end (fn [e]
+                      (if (> (Math/abs (if x? (oget e "translationX") (oget e "translationY")))
+                             constants/drag-threshold)
+                        (do
+                          (anim/animate background-color "rgba(0,0,0,0)")
+                          (anim/animate opacity 0)
+                          (rf/dispatch [:navigate-back]))
+                        (do
+                          #(reset! set-full-height? true)
+                          (anim/animate (if x? pan-x pan-y) 0)
+                          (anim/animate opacity 1)
+                          (anim/animate layout 0)))))))
+
+(defn init-props
```

Comment:
Well, I remember you commenting on this in the composer. I couldn't find a better name and I am open to suggestions :)

It was initially named atoms, now renamed to props, maybe both are equally bad..

> The state scroll-to-index-lock? I would move to init-state

I prefer to keep `reagent` atoms that would, of course, trigger a re-render, and other normal atoms separated. Maybe we can try to come up with a better name.

Reply:
Yeah, naming is hard. As I said, I would give more concrete names. Separating atoms by being reactive or not sometimes works, but often times it doesn't, because this categorization is purely technical, there's not much meaning to it, so at least to me, it makes understanding the code harder. Basically `props` is meaningless when the name is passed down to other functions.

In general, I would (personally) avoid grouping things by generic technical concepts such as reactive or non-reactive, and focus on grouping things by their true purpose in the code.

Reply:
So are you suggesting any changes now? I can put them all in one map `state`. Since we started using that new way or writing functional components, lots of atoms need to be passed down the tree of components, compared to initially where it was one big comp. So I group them to reduce the number of args.

Reply:
It may be too late for this PR, so perhaps my feedback can be better applied for future PRs :) Up to you really. Since you are not satisfied with the suggestions I made it's okay to move on and keep it as is :)

> Since we started using that new way of writing functional components, lots of atoms need to be passed down the tree of components

Yeah, there's room to improve. For this particular lightbox feature though, the complexity seems to be due to the amount of animation values (11), derived values (2) and the design complexities to that force us to do height calculations, so not so much about using functional components.